### ISSUE TITLE
Add --report-file option

### ISSUE THREAD

**Issue by gavv** [gavv/signal-estimator#75]
Add new option `--report-file <path>`.

If it's specified, TextReporter and JsonReporter should write reports to specified file instead of stdout.

By default it should be set to "-", which means write to stdout.

**Comment by PaiktraNom** [comment]
Do you still need this to be looked at? This is my first time contributing to projects

**Comment by gavv** [comment]
Yes. You're welcome, thanks!

**Comment by PaiktraNom** [comment]
Hey @gavv , I'm looking through the repo just to check where the fix could be. I just wanted to check but would the fix be here? https://github.com/gavv/signal-estimator/blob/539b55781e3da4716c791bd702da70c1dc279591/src/cli/Main.cpp#L95

Any input or tips would be useful. Thanks!

**Comment by gavv** [comment]
Yes, this is the place where CLI options are added. CLI options fill Config struct. Then Runner class uses Config and instantiates various objects, including TextReporter and JsonReporter.

**Comment by gavv** [comment]
Also you can take a look how --dump-file option is handled and passed to CsvDumper (you need fresh "main" branch).

**Comment by PaiktraNom** [comment]
I spent the last day trying to understand how all of this interacts. Would it be just this line below, or are there extra things I need to do?
`report_opts ->add_option("--report-file", report_format, "File to report samples (\"-\" for stdout)")`

**Comment by gavv** [comment]
If you mean how to register new option, yes, something like this. See CLI11 docs: https://github.com/CLIUtils/CLI11
